R/translator.R

```diff
@@ -0,0 +1,143 @@
+#' Translator options
+.translator_options <- list(
+  cultural_bignumer_mark = NULL,
+  cultural_punctuation_mark = NULL,
+  cultural_date_format = NULL
+)
+
+#' Translator class
+#'
+#' @field languages character.
+#' @field key_translation character.
+#' @field translation_file character.
+#' @field options list.
+#' @field translations data.frame.
+#' @field translation_language character.
+#'
+#' @return
+#' @export
+#'
+#' @import jsonlite
+#'
+#' @examples
+#' @rdname translator
+Translator <- setRefClass(
+  "Translator",
+  fields = list(
+    languages = "character",
+    key_translation = "character",
+    translation_language = "character",
+    translation_file = "character",
+    options = "list",
+    translations = "data.frame",
+    mode = "character"
+  )
+)
+
+#' @rdname translator
+Translator$methods(
+  initialize = function(translation_file, key_translation) {
+    translation_file <<- translation_file
+    key_translation <<- key_translation
+    options <<- .translator_options
+    # determine the translations format
+    translation_split <- unlist(strsplit(translation_file, "[.]"))
+    translation_format <- translation_split[length(translation_split)]
+    if (length(translation_split) == 1)
+      .read_csv(translation_file, key_translation)
+    else if (translation_format == "json")
+      .read_json(translation_file, key_translation)
+    else
+      stop("Unrecognized format of the file.")
+    # Set default translation language
+    translation_language <<- character(0)
```

Comment:
why are we setting this to `character(0)`? are we checking for this value later?

Reply:
Yes, in the method `translate`.